eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -92,14 +92,14 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
         // Channels are mostly ignored for temporary reasons, likely because they didn't have enough balance to forward
         // the payment. When we're retrying an MPP split, it may make sense to retry those ignored channels because with
         // a different split, they may have enough balance to forward the payment.
-        val (toSend, maxFee) = remainingToSend(nodeParams, d.request, d.pending.values)
+        val (toSend, maxFee) = remainingToSend(d.request, d.pending.values)
         log.debug("retry sending {} with maximum fee {} without ignoring channels ({})", toSend, maxFee, d.ignore.channels.map(_.shortChannelId).mkString(","))
         val routeParams = d.request.routeParams.copy(randomize = true) // we randomize route selection when we retry
         router ! createRouteRequest(nodeParams, toSend, maxFee, routeParams, d, cfg).copy(ignore = d.ignore.emptyChannels())
         retriedFailedChannels = true
         stay() using d.copy(remainingAttempts = (d.remainingAttempts - 1).max(0), ignore = d.ignore.emptyChannels())
       } else {
-        val failure = LocalFailure(Nil, t)
+        val failure = LocalFailure(d.request.totalAmount - d.pending.values.map(_.amount).sum, Nil, t)
```

Comment:
The amount we set here may be inaccurate. I don't think it matters much though.
Here is the scenario where it may be inaccurate:

- we have 3 child payments of 100 sats each
- 1 of them fails: we ask for a new route for an amount of 100 sats
- a second one fails before we receive a response from the router: that will trigger a second route request for 100 sats
- we receive a router error for the first route request: we will over-estimate the amount and count 200 sats in the `LocalFailure`

I think this is ok, as:

- this is an edge case
- the amount in `PaymentFailure` shouldn't be used for business logic, just for reporting/logs

Let me know what you think!

Reply:
We're also missing the total amount, don't you think? Maybe we should define a purpose built `PaymentFailureSummary` class, that could also contain the experiment name, etc. (or even the whole payment config)?

Reply:
Good catch, the total amount would be useful to include.
And why not include the experiment name as well, it doesn't cost anything to add!

Reply:
Added in https://github.com/ACINQ/eclair/pull/1966/commits/66f0a71e2ef8deebc7f680f274fbde554f17ec31